src/ui/zcl_abapgit_gui_router.clas.abap

```diff
@@ -234,16 +234,21 @@ CLASS ZCL_ABAPGIT_GUI_ROUTER IMPLEMENTATION.
 
     DATA: lv_key           TYPE zif_abapgit_persistence=>ty_repo-key,
           lv_last_repo_key TYPE zif_abapgit_persistence=>ty_repo-key,
-          lt_repo_list     TYPE zif_abapgit_persistence=>ty_repos.
+          lt_repo_list     TYPE zif_abapgit_persistence=>ty_repos,
+          lt_repo_obj_list TYPE zif_abapgit_repo_srv=>ty_repo_list.
 
+    FIELD-SYMBOLS: <ls_repo>     LIKE LINE OF lt_repo_list,
+                   <lr_repo_obj> LIKE LINE OF lt_repo_obj_list.
 
     lv_key = ii_event->query( )->get( 'KEY' ).
 
     CASE ii_event->mv_action.
       WHEN zcl_abapgit_gui=>c_action-go_home.
         lv_last_repo_key = zcl_abapgit_persistence_user=>get_instance( )->get_repo_show( ).
-        lt_repo_list = zcl_abapgit_persist_factory=>get_repo( )->list( ).
-
+        lt_repo_obj_list = zcl_abapgit_repo_srv=>get_instance( )->list( ).
+        LOOP AT lt_repo_obj_list ASSIGNING <lr_repo_obj>.
+          APPEND <lr_repo_obj>->ms_data TO lt_repo_list.
```

Comment:
`lt_repo_list` is only used to check if repos exists, the list `lt_repo_obj_list` can be used for this